### DISCUSSION TITLE
2FA with 3rd Party IdP

### DISCUSSION THREAD

**Discussion by cscs-nchalla** [keycloak/keycloak#16633] [Q&A]
Is there any way to use Out of box Authentication steps/flows in Keycloak where We would like to redirect users to third-party identity providers only for 2FA?

**Comment by cgeorgilakis** [comment]
The problem is that any action ( otp, deny access) in Authentication flow is not executed after Identity Provider User log in ( case of Username Password Form, Identity Provider Redirector ). For Keycloak Users everything is ok.

Unfortunately, [my PR](https://github.com/keycloak/keycloak/pull/12891) for fixing is never reviewed by Keycloak Team and no help is provided in order to fix - investigate little test failures. You could see details in [related issue](https://github.com/keycloak/keycloak/issues/10250). Many Keycloak users face the same problem.

You could only configure Identity Provider "Post Login Flow" for an authentication flow per Identity Provider. Of course, this is insufficient for almost all use cases.
